### DISCUSSION TITLE
Berry working size

### DISCUSSION THREAD

**Discussion by trlafleur** [arendst/Tasmota#15956] [Q&A] [answered]
Adding a few new lines of code and many comments, it appears that I may have exceeded Berry's working space... how do I increase the working space??

Removing comments fixes the issue, but it would be nice to retain them...

thanks

**Comment by s-hadinger** [comment]
What do you call working space? Do you mean REPL?

**Reply by sfromis** [reply]
Based on what symptoms do you conclude "may have exceeded Berry's working space"?

If you need more RAM, get a device with PSRAM in addition to the internal ESP32 RAM.

**Reply by trlafleur** [reply]
If I add lots of comments to my code, when I copy it to Berry Console, I get no confirmation of any errors, but the program did not run, remove comments, and works just fine... it appears that I have reached some limit in Berry.

**Reply by sfromis** [reply]
Copy/pasting into the REPL is probably not a great plan for long pieces of code, seems to get into trouble when getting above approx 14KB. With PSRAM available, you can go higher, according to a brief testing....

**Comment by s-hadinger** [comment, marked as answer]
The web REPL requires to pass a large text buffer and then to compile it in place. If it gets too large you need PSRAM. 

Loading from a file is much more frugal since it is done in streaming mode. You don't have such limits with `load()`

**Comment by trlafleur** [comment]
thanks, that is the issue... file is large...

~~ _/) ~~~~ _/) ~~~~ _/) ~~~~ _/) ~~

Tom Lafleur
